gr-blocks/lib/file_source_impl.cc

```diff
@@ -110,6 +110,16 @@ namespace gr {
 	throw std::runtime_error("can't open file");
       }
 
+			//Check to ensure the file will be consumed according to item size
+			fseek(d_new_fp, 0, SEEK_END);
+			int file_size = ftell(d_new_fp);
+			rewind (d_new_fp);
+
+			//Warn the user if part of the file will not be consumed.
+			if(file_size % d_itemsize){
+				printf("WARNING: File will not be fully consumed with current output type of size %d bytes.\n", d_itemsize);
```

Comment:
Also, your tab/space settings are off. As shown in the test of the code, we use 2 spaces per scope indent, no tabs.